carbonmark-api/test/flows/authentication.test.ts

```diff
@@ -0,0 +1,52 @@
+import { ethers } from "ethers";
+import { FastifyInstance } from "fastify";
+import { build } from "../helper";
+import { DEV_URL } from "../test.constants";
+
+// The private key of the account to sign the message with
+const MOCK_PRIVATE_KEY =
+  "0x0123456789012345678901234567890123456789012345678901234567890123";
+
+// Assuming you have a connected provider
+const wallet = new ethers.Wallet(MOCK_PRIVATE_KEY);
+
+describe("Authentication flow", () => {
+  let app: FastifyInstance;
+
+  // Setup the server
+  afterEach(async () => await app.close());
+  beforeAll(async () => {
+    app = await build();
+  });
+
+  test("should authenticate user", async () => {
```

Comment:
nice nice you mind throwing in the sad path too while you are at it? This is a pretty important test :D

Reply:
Good idea, done